src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -1735,6 +1764,18 @@ public void SetParameterNames(IEnumerable<string> parameters)
             _parameterIndexToNameMap = parameterIndexToNameMap;
         }
 
+        public void SetVariableTypeIndex(TypeDesc td, uint index)
+        {
+            uint typeIndex = 0;
+            if (td.IsPrimitive)
+            {
+                typeIndex = TypesDebugInfo.PrimitiveTypeDescriptor.GetPrimitiveTypeIndex(td);               
+            }
+            // Add indexes one by one.
+            Debug.Assert(_variableToTypeIndex.Count == index);
```

Comment:
Nit: It feels like the `index` variable is just a leftover from when `_variableToTypeIndex` was just a dictionary. Maybe we can get rid of it an use `_variableToTypeIndex.Count`?